packages/native/types/base.d.ts

```diff
@@ -89,21 +87,26 @@ export interface StyledOptions<Props> {
   shouldForwardProp?(propName: PropertyKey): boolean
 }
 
+export interface StyledWithComponent<ComponentProps extends {}> {
+  withComponent<
+    Component extends ComponentType<ComponentPropsWithoutRef<Component>>
+  >(
+    component: Component
+  ): StyledComponent<ComponentProps & ComponentPropsWithoutRef<Component>>
+  withComponent<ComponentName extends ReactNativeComponentNames>(
+    component: ReactNativeComponents[ComponentName]
+  ): StyledComponent<ComponentProps, ReactNativeComponentProps<ComponentName>>
+}
+
 /**
  * @typeparam ComponentProps  Props which will be included when withComponent is called
  * @typeparam SpecificComponentProps  Props which will *not* be included when withComponent is called
  */
-export interface StyledComponent<
+export type StyledComponent<
   ComponentProps extends {},
   SpecificComponentProps extends {} = {}
-> extends React.FC<ComponentProps & SpecificComponentProps> {
-  withComponent<C extends React.ComponentType<React.ComponentProps<C>>>(
-    component: C
-  ): StyledComponent<ComponentProps & PropsOf<C>>
-  withComponent<Tag extends keyof ReactNativeElements>(
-    tag: Tag
-  ): StyledComponent<ComponentProps, ReactNativeElements[Tag]>
-}
+> = ComponentType<ComponentProps & SpecificComponentProps> &
+  StyledWithComponent<ComponentProps>
```

Comment:
It doesn't make much sense to type `StyledComponent` as an extension to a functional component. Technically it should be typed as a `ForwardRefExoticComponent` (and more generally, `NamedExoticComponent`). This might be a more appropriate choice (i'll give it a try in a moment).

Reply:
`NamedExoticComponent` works appropriately here so i've gone with that choice.

Reply:
this makes sense, cc @JakeGinnivan - shouldn't we do the same for web types?